src/components/menu/MenuTopLevelControlled.tsx

```diff
@@ -0,0 +1,68 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import { OpenAction, IMenuPositionConfig } from './MenuSharedTypes';
+import MenuOpener from './MenuOpener/MenuOpener';
+
+export interface IMenuTopLevelControlledProps extends Partial<IMenuPositionConfig> {
```

Comment:
The type with `Partial<IMenuPositionConfig>` seems quite vague to me as I never know which props from `IMenuPositionConfig` are expected here. I'd prefer listing all expected props here. Not just here but everywhere, where `Partial<IMenuPositionConfig>` occurs

Reply:
I also encountered this problem and I think that it is much better than it used to be. I cleaned it up and now any component that uses `Partial<>` in props is actually using **all** the props in that type and at the same time it has them **all** optional.

Reply:
That makes sense. Thanks for the explanation.